### ISSUE TITLE
Using Named Parameters for Exceptions CS0121

### ISSUE THREAD

**Issue by kbrashears5** [NLog/NLog#4209]
**NLog version**: (e.g. 4.4.13)
4.7.6

**Platform**: .Net 3.5 / .Net 4.0 / .Net 4.5 / Mono 4 / Silverlight 4 / Silverlight 5 / Xamarin Android / Xamarin iOs / .NET Core 1 / .NET Core 2 / .NET Core 3
.Net 4.8

- What is the current result?
Using named parameters when trying to log exceptions produces compile error:
```c#
this.Logger.Error(exception: ex, message: string.Empty);
```

- What is the expected result?
No error to be thrown - why are there two different method signatures for the same thing?
- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
N/A
- Please post full exception details (message, stacktrace, inner exceptions)
> Error	CS0121	The call is ambiguous between the following methods or properties: 'ILogger.Error(Exception, string)' and 'ILogger.Error(string, Exception)'
- Are there any workarounds? yes/no
not use named parameters
- Is there a version in which it did work? 
this is the first version I've tried
- Can you help us by writing an unit test?
N/A

**Comment by snakefoot** [comment]
This will be resolved with NLog 5.0 where this obsolete method is removed:

```c#
        [Obsolete("Use Error(Exception exception, string message, params object[] args) method instead.")]
        public void Error([Localizable(false)] string message, Exception exception)
```

**Comment by 304NotModified** [comment]
I assume your question has been answered, if not, please let us know!
